include/support.h

```diff
@@ -218,6 +218,9 @@ inline bool is_empty(const char *str) noexcept
 // Scans the provided command-line string for the '/'flag and removes it from
 // the string, returning if the flag was found and removed.
 bool ScanCMDBool(char *cmd, const char * flag);
+
+// No-return variant of ScanCMDBool when the flag's presence is not important
+void maybe_remove_flag(char *cmd, const char *flag);
```

Comment:
I don't like the naming here. It bears no relationship to `ScanCMDBool`, despite the fact that it is what it is calling. Also, people might think you could use it for flags with values, when you can't.

Reply:
Agreed.

 My dillema was that the original name wasn't forthcoming as to what it actually did - and only after reading it's code, I then came up with a clearer name. 

ScanCMDBool also doesn't fit with our case rules either..so I'm hesitant to pile on another poorly named function that also doesn't fit with our recommended naming rules. 

The original function is only used in shell_cmd too - so it can be lifted out of support.

I don't want future readers to go through what I did to figure out what it's doing: so, proposal:

 - move it into the shell_cmd as a local static.
 - rename it to something that says "it will find and remove non-paramaterized slash-flags from the provided string". 
 - then we'll name the no-return variant using that, and it will also be local static in shell_cmd. 

Trying to come up with a better name: what's a name for a command line flag that doesn't have a value?

Reply:
Hmm, maybe we should look at the entire suite of `ScanCMD` functions. It looks like they're all only used in `shell_cmds`. I feel that would be better suited for a separate PR though.

So I think you should probably drop `maybe_remove_flag`, add the void casts at the `ScanCMDBool` calls where necessary, and we can revisit it when looking at the entire `ScanCMD` suite.

Reply:
Fully agree. 
This path will lead to less technical debt (from a naming / readability perspective) more than a piecemeal approach here.